config/routes.rb

```diff
@@ -0,0 +1,17 @@
+Rails.application.routes.draw do
+
+  root 'pages#index'
+
+  resources :works
+  post '/works/:id/upvote', to: 'works#upvote', as: 'upvote'
+
+  resources :users, only: [:index, :show]
+
+  resources :votes, only: [:new, :create]
```

Comment:
Do you need a form or a `create` action for `votes`?  Wouldn't `upvote` do just fine?